pkgs/development/tools/analysis/ikos/default.nix

```diff
@@ -0,0 +1,31 @@
+{ stdenv, lib, fetchFromGitHub, cmake, boost
+, gmp, llvmPackages_7, sqlite, python3
+, ocamlPackages, mpfr, ppl, doxygen, graphviz
+}:
+
+llvmPackages_7.stdenv.mkDerivation rec {
+  name = "ikos";
+  version = "2.1";
+
+  src = fetchFromGitHub {
+    owner = "NASA-SW-VnV";
+    repo = name;
+    rev = "v${version}";
+    sha256 = "09nf47hpk5w5az4c0hcr5hhwvpz8zg1byyg185542cpzbq1xj8cb";
+  };
+
+  python = python3.withPackages (ps: [ ps.pygments ]);
```

Comment:
I'd prefer this in a `let` rather than in the `mkDerivation` arguments. I got confused reading it wondering why `mkDerivation` would need a `python`, before I realised it was just here so it could be used as a build input.

Reply:
Example of the `let` version: https://github.com/NixOS/nixpkgs/blob/ad3e9191d16722ea3eec32f4cd689eea730f39f6/pkgs/tools/inputmethods/ibus-engines/ibus-typing-booster/default.nix#L5-L12